adl/dconfig.adl

```diff
@@ -0,0 +1,51 @@
+module dconfig {
+  import types.StringKeyMap;
+  import types.ConfigName;
+  import types.FilePath;
+  import types.S3Path;
+  import sys.types.Set;
+
+  /// ConfigName for switchable dynamic configs - eg 'queue' 'processing'
+  type DynamicConfigName = ConfigName;
+
+  /// Name of a mode of a dynamic config - eg 'active' 'inactive'
+  type DynamicConfigMode = String;
+
+  /// A config source that can be changed to different named Modes at runtime.
+  struct DynamicJsonSource {
+    DynamicConfigMode defaultMode;
+
+    StringKeyMap<DynamicConfigMode, JsonSource> modes;
+  };
+
+  type DynamicConfigNameModeMap = StringKeyMap<DynamicConfigName, DynamicConfigMode>;
+  type DynamicConfigNameJSrcMap = StringKeyMap<DynamicConfigName, DynamicJsonSource>;
+
+  /// Listing of available modes per configName
+  type DynamicConfigOptions = StringKeyMap<DynamicConfigName, Set<DynamicConfigMode>>;
```

Comment:
What ADL uses this definition?

Reply:
`DynamicConfigOptions` also only used internally in haskell for the implementation.